parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -100,58 +100,40 @@ def parse_requests(self):
                     LOGGER.debug('{} tree {} vs {}, semver {} vs {}'.format(
-                        key, old_module.get('tree-type'), val.get('tree-type'),
-                        old_module.get('derived-semantic-version'), val.get('derived-semantic-version'))
+                        key, old_module.get('tree-type'), new_module.get('tree-type'),
+                        old_module.get('derived-semantic-version'), new_module.get('derived-semantic-version'))
                     )
-                    if ret_modules.get(key) is None:
-                        ret_modules[key] = val
-                    else:
-                        ret_modules[key]['name'] = val['name']
-                        ret_modules[key]['revision'] = val['revision']
-                        ret_modules[key]['organization'] = val['organization']
-                        ret_modules[key]['tree-type'] = val.get('tree-type')
-                        ret_modules[key]['derived-semantic-version'] = val.get('derived-semantic-version')
-                if val.get('dependents') is not None and len(val.get('dependents')) != 0:
+                    ret_modules[key] = new_module
+                if new_module.get('dependents'):
                     if old_module.get('dependents') is None:
                         if ret_modules.get(key) is None:
-                            ret_modules[key] = val
-                            LOGGER.debug('dependents {} vs {}'.format(None, val['dependents']))
-                            continue
-                        ret_modules[key]['name'] = val['name']
-                        ret_modules[key]['revision'] = val['revision']
-                        ret_modules[key]['organization'] = val['organization']
-                        ret_modules[key]['dependents'] = val['dependents']
-                        LOGGER.debug('dependents {} vs {}'.format(None, val['dependents']))
+                            ret_modules[key] = new_module
+                        LOGGER.debug('dependents {} vs {}'.format(None, new_module['dependents']))
                         continue
-                    for dep in val['dependents']:
-                        found = False
+                    for new_dep in new_module['dependents']:
                         for old_dep in old_module['dependents']:
-                            if (dep.get('name') == old_dep.get('name') or
-                                    dep.get('revision') == old_dep.get('revision') or
-                                    dep.get('schema') == old_dep.get('schema')):
-                                found = True
+                            if (new_dep.get('name') == old_dep.get('name') and
+                                    new_dep.get('revision') == old_dep.get('revision') and
+                                    new_dep.get('schema') == old_dep.get('schema')):
                                 break
-                        if not found:
+                        else:
```

Comment:
This **else** is either redundant or has a bad indentation. Please check it.

Reply:
The else is intentional, it runs if the for loop finished without breaking.